Feature/zenko 3735 kafka dashboard

lavigne958 opened the pull request:
**What does this PR do, and why do we need it?**
This PR introduces 2 new dashboards for grafana

**Which issue does this PR fix?**
None, it is an improvement.

Commit:
ZENKO-3735 - add first kafka dashboard - system metrics

Commit:
ZENKO-3735 - add second kafka dashboard - kafka metrics

bert-e commented:
# Hello lavigne958,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/zenko/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

francoisferrand commented:
can we/should we merge both into a single Kafka dashboard ?

lavigne958 commented:
> can we/should we merge both into a single Kafka dashboard ?

yes, but that's gonna be a BIG dashboard.

francoisferrand commented:
It may be a bit bigger, but the second dashboard is quite small, and there are duplicated metrics between both (jvm_memory_bytes_used for exemple, though this is not an exaustive list : just the first one i looked...). On top of that, it may be easier for user to know where to look (only one dashboard) and use (with the ability to view multiple metrics in correlation). To make the size manageable, there may be "sections" which can be collapsed as needed.

My opinion on this is that (generally) we should have separate dashboard only if this is required for 2 independent purposes, and avoid having too many dashboards which would IMHO bring more confusion than help... I am not sure if this is the case here, but you should definitly check with @ImaneSAniss and/or @anurag4DSB (or possibly some in-house kafka experts?

francoisferrand commented on monitoring/dashboards/kafka_system.json:
```diff
@@ -0,0 +1,588 @@
+  "version": 1,
+  "links": [],
+  "gnetId": 721,
+  "description": "Kafka resource usage and throughput"
```
throughput is not a "system" metrics (as in the title of diagram), and the associated metrics (`kafka_server_brokertopicmetrics_bytes*_total`) seems to be in the other dashboard as well.

lavigne958 replied:
done, I removed the first dashboard, in fact all its metrics are already in the second dashboard.

Commit:
ZENKO-3735 - remove duplicate dashboard

the first dashboard contains only metrics that already exists
in the second dashboard, simply remove the dashboard

miniscruff commented:
Can you include screenshots for dashboard PRs? Reviewing raw JSON dashboards can be quite hard.

lavigne958 commented:
> Can you include screenshots for dashboard PRs? Reviewing raw JSON dashboards can be quite hard.

Sure here it is.

For the context we discussed the following points for the dashboards, you can see the README in[ jordi's PR](https://github.com/scality/Zenko/pull/1405).

The screenshots:
![image](https://user-images.githubusercontent.com/15084254/143409216-f22903c0-f07e-4feb-9d0c-39c15f7f4f62.png)
![image](https://user-images.githubusercontent.com/15084254/143409289-e1347640-c974-412f-bd9f-109439e0f80f.png)
![image](https://user-images.githubusercontent.com/15084254/143409351-e1b0282a-5b37-4a83-9234-4d01da578212.png)
![image](https://user-images.githubusercontent.com/15084254/143409433-32b24373-2c2d-447c-b430-df61902fed1c.png)
![image](https://user-images.githubusercontent.com/15084254/143409501-26bb8b48-4e8e-4508-9610-4d0c5479eaef.png)
![image](https://user-images.githubusercontent.com/15084254/143409548-ed36d77b-5d27-47ab-be4e-88514aa734e9.png)
![image](https://user-images.githubusercontent.com/15084254/143409588-d4c1494f-02f7-44e1-898e-0804e4d9125d.png)
![image](https://user-images.githubusercontent.com/15084254/143409699-fdb46f21-9682-4fda-9a24-9f3fc894f310.png)
![image](https://user-images.githubusercontent.com/15084254/143409740-0c32062f-1285-4298-9cb4-24d73892d7d4.png)

lavigne958 commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.1`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/0.9`
* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.1`



Please check the status of the associated issue ZENKO-3735.

Goodbye lavigne958.
